cmd/admin-handlers.go

```diff
@@ -244,39 +245,111 @@ func (a adminAPIHandlers) ServerInfoHandler(w http.ResponseWriter, r *http.Reque
 	if objectAPI == nil {
 		return
 	}
+	var jsonBytes []byte
+	var err error
+
+	vars := mux.Vars(r)
+	switch infoType := vars["infoType"]; infoType {
+
+	case "vault":
+		if GlobalKMS == nil {
+			writeErrorResponseJSON(ctx, w, errorCodes.ToAPIErr(ErrKMSNotConfigured), r.URL)
+			return
+		}
+		config, err := readServerConfig(ctx, objectAPI)
+		if err != nil {
+			writeErrorResponseJSON(ctx, w, toAdminAPIErr(ctx, err), r.URL)
+			return
+		}
+		var Environment = environment{}
+		endPoint := Environment.Get(EnvVaultEndpoint, config.KMS.Vault.Endpoint)
+		vaultName := Environment.Get(EnvVaultKeyName, config.KMS.Vault.Key.Name)
+		vaultType := Environment.Get(EnvVaultAuthType, config.KMS.Vault.Auth.Type)
+
+		var perm = Permission{}
+		kmsContext := crypto.Context{"MinIO admin API": "KMSKeyStatusHandler"} // Context for a test key operation
+		// Generate a new key using the KMS.
+		_, sealedKey, e := GlobalKMS.GenerateKey(vaultName, kmsContext)
+		if e == nil {
+			perm.Encryption = "OK"
+		} else {
+			perm.Encryption = "FAIL"
+		}
+
+		// Verify that we can indeed decrypt the (encrypted) key
+		_, e = GlobalKMS.UnsealKey(vaultName, sealedKey, kmsContext)
+		if e == nil {
+			perm.Decryption = "OK"
+		} else {
+			perm.Decryption = "FAIL"
+		}
 
-	serverInfo := globalNotificationSys.ServerInfo(ctx)
-	// Once we have received all the ServerInfo from peers
-	// add the local peer server info as well.
-	serverInfo = append(serverInfo, ServerInfo{
-		Addr: getHostName(r),
-		Data: &ServerInfoData{
-			StorageInfo: objectAPI.StorageInfo(ctx),
-			ConnStats:   globalConnStats.toServerConnStats(),
-			HTTPStats:   globalHTTPStats.toServerHTTPStats(),
-			Properties: ServerProperties{
-				Uptime:       UTCNow().Sub(globalBootTime),
-				Version:      Version,
-				CommitID:     CommitID,
-				DeploymentID: globalDeploymentID,
-				SQSARN:       globalNotificationSys.GetARNList(),
-				Region:       globalServerConfig.GetRegion(),
+		vaultInfo := VaultInfo{
+			Endpoint: endPoint,
+			Name:     vaultName,
+			Type:     vaultType,
+			Perm: Permission{
```

Comment:
This can be simply, `Perm: perm`, right?

Reply:
thanks, corrected it.